### ISSUE TITLE
Bug: [prefer-nullish-coalescing] Optional chain produces false positives

### ISSUE THREAD

**Issue by steverep** [typescript-eslint/typescript-eslint#8648]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.2&fileType=.ts&code=JYOwLgpgTgZghgYwgAgOoHsoBNkG8CwAUMsgM5wCeAXGWFKAOYDcRAvkUQuiKWGQA4Q4Aa2QBeZAAoA7piw0M2ZAB9kIAK4AbTQEpxAPjxESXHuk0QAdJvQMZcgPyXyFFaoBEAOXRgAFo2QwdDJKdx02IiA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYkACAXAngA4oDG0AlobgLQrzkB2uA9IdIgGaLTUML3IAFtRIB7AIZJkJRgHN0UbtFHRIAXxBqgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
interface Word {
  say: string;
}

const speak = (word: Word | null) => {
  console.log(word?.say || "Nothing to say")
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {"@typescript-eslint/prefer-nullish-coalescing": "error"}
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

There should be no error because the `??` operator will produce a different result if  `say` is an empty string.

### Actual Result

Rule is triggered and recommends switch to `??`.

**Comment by auvred** [comment]
I guess `"ignorePrimitives": {"string": true}` option should do the trick

Docs: https://typescript-eslint.io/rules/prefer-nullish-coalescing/#ignoreprimitives

**Comment by steverep** [comment]
> I guess `"ignorePrimitives": {"string": true}` option should do the trick

Sure, that resolves the error, but why should I have to do that?  The default for this rule under the `stylistic-type-checked` shared config is to error with `ignorePrimitives` as `false`, which seems like a big overreach to me considering optional chain situations like this.  The docs for stylistic rules say:

> Rules considered to be best practice for modern TypeScript codebases, but that do not impact program logic.

Clearly this rule can easily impact program logic.  The docs for the `ignoreConditionalTests` and `ignoreMixedLogicalExpressions` admit as much by stating the switch from `||` to `??` "could cause bugs", yet they are still false by default.  By contrast, the `ignorePrimitives` option makes no such statement, and it's not clear the optional chain behavior was considered.  Meaning the docs talk about the primitive being directly mixed with `null` or `undefined`, which is not the case in this example (i.e. the logical operand can be `undefined` but the `say` property cannot).

IMHO, this is a dangerous rule in its current state, especially considering it will make changes with `eslint --fix` which is commonly run automatically on save or precommit hooks.

**Comment by steverep** [comment]
@bradzacher I don't understand closing this as working as intended.  The default config for this rule effectively renders it impossible to use an optional chain, which ends in a primitive, as the left operand of `||`.  Is that really the rule's intention?

Per your [comment on #7628](https://github.com/typescript-eslint/typescript-eslint/issues/7628#issuecomment-1721883811) (which remains open despite having a very similar premise):

> The other problem is that once you remove primitive cases you are pretty sparse on the remaining safe cases.

And yet the recommended solution here is to set `ignorePrimitives: true` here even though the property types are not the issue?
